src/notification/index.js

```diff
@@ -71,23 +71,44 @@ export const getAccountFromNotificationData = (
     return null;
   }
 
-  if (urlMatches.length > 1) {
-    // The user has several accounts in the notification's realm.  We should
-    // be able to tell the right one using the notification's `user_id`...
-    // except we don't store user IDs in `accounts`, only emails.  Until we
-    // fix that, just ignore the information.
-    logging.warn('notification realm_uri ambiguous; multiple matches found', {
-      realm_uri,
-      parsed_url: realmUrl,
-      match_count: urlMatches.length,
-      unique_identities_count: new Set(urlMatches.map(matchIndex => identities[matchIndex].email))
-        .size,
-    });
-    // TODO get user_id into accounts data, and use that
-    return null;
+  // TODO(server-2.1): Remove this, because user_id will always be present
+  if (user_id === undefined) {
+    if (urlMatches.length > 1) {
+      logging.warn(
+        'notification realm_uri ambiguous; multiple matches found; user_id missing (old server)',
+        {
+          realm_uri,
+          parsed_url: realmUrl,
+          match_count: urlMatches.length,
+          unique_identities_count: new Set(urlMatches.map(matchIndex => accounts[matchIndex].email))
+            .size,
+        },
+      );
+      return null;
+    } else {
+      return urlMatches[0];
+    }
   }
 
-  return urlMatches[0];
+  // There may be multiple accounts in the notification's realm. Pick one
+  // based on the notification's `user_id`.
+  const userMatch = urlMatches.find(urlMatch => {
+    const { userId } = accounts[urlMatch];
+    return userId !== null && userId === user_id;
```

Comment:
This `userId !== null` is always true if `userId === user_id` is, right? So it can be dropped, and this function simplifies back down to just returning an expression.